src/fetchcode/package_util.py

```diff
@@ -723,3 +730,294 @@ def get_package_info(cls, gh_purl, package_name):
+                if data_list[0] == cocoapods_org_pod_name:
+                    data_list.pop(0)
+                sorted_data_list = sorted(
+                    data_list,
+                    key=lambda x: versions.SemverVersion(x),
+                    reverse=True,
+                )
+                return sorted_data_list
+        return None
+    except:
+        print(f"Error retrieving cocoapods tag data from cdn.cocoapods.org")
+        return None
+
+
+def construct_cocoapods_package(
+    purl,
+    name,
+    hashed_path,
+    repository_homepage_url,
+    cocoapods_org_gh_repo_owner,
+    cocoapods_org_gh_repo_name,
+    tag,
+    cocoapods_org_pod_name
+):
+    name = name
+    homepage_url = None
+    vcs_url = None
+    github_url = None
+    bug_tracking_url = None
+    code_view_url = None
+    license_data = None
+    declared_license = None
+    primary_language = None
+
+    if cocoapods_org_gh_repo_owner and cocoapods_org_gh_repo_name:
+        name = cocoapods_org_gh_repo_name
+        namespace = cocoapods_org_gh_repo_owner
+        base_path = "https://api.github.com/repos"
+        api_url = f"{base_path}/{namespace}/{name}"
+        gh_repo_api_response = utils.get_github_rest_no_exception(api_url)
+
+        if "Failed to fetch" not in gh_repo_api_response:
+            homepage_url = gh_repo_api_response.get("homepage")
+            vcs_url = gh_repo_api_response.get("git_url")
+            license_data = gh_repo_api_response.get("license") or {}
+            declared_license = license_data.get("spdx_id")
+            primary_language = gh_repo_api_response.get("language")
+
+        github_url = "https://github.com"
+        bug_tracking_url = f"{github_url}/{namespace}/{name}/issues"
+        code_view_url = f"{github_url}/{namespace}/{name}"
+
+    corrected_name = cocoapods_org_pod_name
+    podspec_api_url = f"https://raw.githubusercontent.com/CocoaPods/Specs/master/Specs/{hashed_path}/{corrected_name}/{tag}/{corrected_name}.podspec.json"
+    podspec_api_response = utils.get_json_response(podspec_api_url)
+
+    if "Failed to fetch" in podspec_api_response:
+        logger.error(f"{podspec_api_response}")
+        print(f"{podspec_api_response}")
+        return
+
+    homepage_url = podspec_api_response.get("homepage")
+
+    lic = podspec_api_response.get("license")
+    extracted_license_statement = None
+    if isinstance(lic, dict):
+        extracted_license_statement = lic
+    else:
+        extracted_license_statement = lic
+    if not declared_license:
+        declared_license = extracted_license_statement
+
+    source = podspec_api_response.get("source")
+    vcs_url = None
+    download_url = None
+    if isinstance(source, dict):
+        git_url = source.get("git", "")
+        http_url = source.get("http", "")
+        if http_url:
+            download_url = http_url
+        if git_url and not http_url:
+            if git_url.endswith(".git") and "github" in git_url:
```

Comment:
`"github" in git_url` is not a reliable way for identifying GitHub URLs.

```suggestion
            if git_url.endswith(".git") and git_url.startswith("https://github.com/"):
```

Reply:
Much better -- thanks @keshav-space .  Done.